internal/provider/resource_vault_cluster.go

```diff
@@ -1031,6 +1078,30 @@ func flattenObservabilityConfig(config *vaultmodels.HashicorpCloudVault20201125O
 			}
 		}
 	}
+
+	if cloudwatch := config.Cloudwatch; cloudwatch != nil {
+		configMap["cloudwatch_access_key_id"] = cloudwatch.AccessKeyID
+		configMap["cloudwatch_region"] = cloudwatch.Region
+		// ensure we only set properties that are defined in metrics/audit-logs streaming
+		if propertyName == "metrics_config" {
+			// Namespace is only used for streaming metrics
+			configMap["cloudwatch_namespace"] = cloudwatch.Namespace
+		} else {
```

Comment:
I noticed that a field called `dataset` is defined on the `config.ElasticSearch` struct. Would that be initialized here as well? Is it only defined when streaming metrics similar to the `cloudwatch_namespace` from earlier?